addons/product/models/product_template.py

```diff
@@ -993,6 +993,11 @@ def _get_closest_possible_combinations(self, combination):
             elements as possible from the given combination.
         :rtype: generator of recordset of product.template.attribute.value
         """
+        # for performance, avoid looping if the current combination is possible
+        combination = combination or self.env['product.template.attribute.value']
+        if self._is_combination_possible(combination):
+            yield(combination)
+            return _("There are no remaining closest combination.")
```

Comment:
@seb-odoo `combinations` instead of `combination`.
Also needs to be added to the source terms (`.pot` file).

Reply:
It's the same string as 10 lines below.